fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -238,6 +277,8 @@ public List<String> listPartitionNames(String dbName, String tblName) {
             StructLike partition = fileScanTask.file().partition();
             partitionNames.add(convertIcebergPartitionToPartitionName(icebergTable.spec(), partition));
         }
+
+        latestTable.setCachedPartitionNames(partitionNames);
```

Comment:
set the `cachedSnapshotId` first and then set `cachedPartitionNames` would lead to concurrent cache update ?

Reply:
scan task used snapshot which id is cachedSnapshotId, so cachedPartitionNames would be same for same  cachedSnapshotId